File: Diameter.py
def find_diameter(root):
    if root is None:
        return 0

    left_height = find_height(root.left)
    right_height = find_height(root.right)

    left_diameter = find_diameter(root.left)
    right_diameter = find_diameter(root.right)

    return max(left_diameter, right_diameter, left_height + right_height + 1)

File: Java/Diameter.java
public int findDiameter(Node root) {
    if (root == null)
        return 0;

    int leftHeight = findHeight(root.left);
    int rightHeight = findHeight(root.right);

    int leftDiameter = findDiameter(root.left);
    int rightDiameter = findDiameter(root.right);

    return Math.max(Math.max(leftDiameter, rightDiameter), leftHeight + rightHeight + 1);
}
